### ISSUE TITLE
clabel raises KeyError with level on boundary since matplotlib 2.1.0

### ISSUE THREAD

**Issue by jschueller** [matplotlib/matplotlib#9742]
As of latest matplotlib 2.1.0 clabel raises a KeyError when the level is on the boundary of z values

Here I try plt.contour on null values except for one, with levels=[1.0] so it lies on the boundary (it's ok with levels=[1.0-1e-6])
the result of plt.contour is obviously empty, but now the plt.clabel call fails:

```python
import matplotlib
import numpy as np
import matplotlib.cm as cm
import matplotlib.mlab as mlab
import matplotlib.pyplot as plt

levels = [1.0]
delta = 0.025
x = np.arange(-3.0, 3.0, delta)
y = np.arange(-2.0, 2.0, delta)
X, Y = np.meshgrid(x, y)
Z1 = mlab.bivariate_normal(X, Y, 1.0, 1.0, 0.0, 0.0)
Z = 0.0 * Z1
Z[0,0] = 1.0

plt.figure()
CS = plt.contour(X, Y, Z, levels=levels)
fmt = {}
for l in levels:
    fmt[l] = 'w='+str(l)
plt.clabel(CS, inline=1, fontsize=10, fmt=fmt)
plt.title('Simplest default with labels')

plt.show()
```

traceback:
```
KeyError                                  Traceback (most recent call last)
/home/schueller/projects/openturns/contour_demo.py in <module>()
     21 for l in levels:
     22     fmt[l] = 'w='+str(l)
---> 23 plt.clabel(CS, inline=1, fontsize=10, fmt=fmt)
     24 plt.title('Simplest default with labels')
     25 

/home/schueller/.local/lib/python3.5/site-packages/matplotlib/pyplot.py in clabel(CS, *args, **kwargs)
   2753                       mplDeprecation)
   2754     try:
-> 2755         ret = ax.clabel(CS, *args, **kwargs)
   2756     finally:
   2757         ax._hold = washold

/home/schueller/.local/lib/python3.5/site-packages/matplotlib/axes/_axes.py in clabel(self, CS, *args, **kwargs)
   5880 
   5881     def clabel(self, CS, *args, **kwargs):
-> 5882         return CS.clabel(*args, **kwargs)
   5883     clabel.__doc__ = mcontour.ContourSet.clabel.__doc__
   5884 

/home/schueller/.local/lib/python3.5/site-packages/matplotlib/contour.py in clabel(self, *args, **kwargs)
    210             blocking_contour_labeler(inline, inline_spacing)
    211         else:
--> 212             self.labels(inline, inline_spacing)
    213 
    214         # Hold on to some old attribute names.  These are deprecated and will

/home/schueller/.local/lib/python3.5/site-packages/matplotlib/contour.py in labels(self, inline, inline_spacing)
    637             con = self.collections[icon]
    638             trans = con.get_transform()
--> 639             lw = self.get_label_width(lev, self.labelFmt, fsize)
    640             lw *= self.ax.figure.dpi / 72.0  # scale to screen coordinates
    641             additions = []

/home/schueller/.local/lib/python3.5/site-packages/matplotlib/contour.py in get_label_width(self, lev, fmt, fsize)
    264         """
    265         if not isinstance(lev, six.string_types):
--> 266             lev = self.get_text(lev, fmt)
    267 
    268         lev, ismath = text.Text.is_math_text(lev)

/home/schueller/.local/lib/python3.5/site-packages/matplotlib/contour.py in get_text(self, lev, fmt)
    327                 except:
    328                     warnings.warn('deeeeeeeeeeeeeeeeeeeeeeeeeeeee'+str(fmt))
--> 329                 return fmt[lev]
    330             elif callable(fmt):
    331                 return fmt(lev)

KeyError: 0.0
```

I think it should rather warn like contour does: warnings.warn("No contour levels were found

I'm on Matplotlib 2.1.0 from pip, Ubuntu 16.04, backend TkAgg, Python 3.5, reproducible on Trusty circleci containers

**Comment by afvincent** [comment]
FWIW, using the MWE above to bisect, the behavior changed with 6ac197f:
```git
6ac197fd58404633499c3fb7bacc461c3c1666f9 is the first bad commit
commit 6ac197fd58404633499c3fb7bacc461c3c1666f9
Author: Eric Firing <efiring@hawaii.edu>
Date:   Sun Jun 4 11:09:24 2017 -1000

    BUG: handle empty levels array in contour, closes #7486
    
    This relates to line contours only, not to filled contours.
    
    This handles the case where an empty array is provided as the
    "levels" kwarg, or where for any reason there are no levels within
    the data range.  An example is the case of uniform z. The
    levels array is then set to the minimum value of z so that
    the plotting can proceed, and a warning is generated.
```

Attn @efiring?

**Comment by jklymak** [comment]
Yeah. The problem is that there is a contour level passed to `clabel`.  In this case its value is `0` because of 6ac197f setting a level to the lowest data value.  Not usually a problem, except the format dictionary you have provided `fmt` doesn't have an entry for level `0` and the labelling fails.  

I don't see why we wouldn't check the user's dictionary for the existence of the level and that seems to fix the bug

**Comment by jklymak** [comment]
Fixed by #9743.  Thanks for the helpful report!
